include/godot_cpp/classes/wrapped.hpp

```diff
@@ -132,16 +136,41 @@ struct EngineClassRegistration {
 } // namespace godot
 
 #ifdef HOT_RELOAD_ENABLED
-#define _GDCLASS_RECREATE(m_class, m_inherits)                                                   \
+#define _GDCLASS_RECREATE(m_class)                                                               \
 	m_class *new_instance = (m_class *)memalloc(sizeof(m_class));                                \
 	Wrapped::RecreateInstance recreate_data = { new_instance, obj, Wrapped::recreate_instance }; \
 	Wrapped::recreate_instance = &recreate_data;                                                 \
 	memnew_placement(new_instance, m_class);                                                     \
 	return new_instance;
 #else
-#define _GDCLASS_RECREATE(m_class, m_inherits) return nullptr;
+#define _GDCLASS_RECREATE(m_class) return nullptr;
 #endif
 
+namespace godot {
+
+template <class T>
+class ClassCreator {
+public:
+	static GDExtensionObjectPtr create(void *data) {
+		if constexpr (!std::is_abstract_v<T>) {
+			T *new_object = memnew(T);
+			return new_object->_owner;
+		} else {
+			return nullptr;
+		}
+	};
+
+	static GDExtensionClassInstancePtr recreate(void *data, GDExtensionObjectPtr obj) {
+		if constexpr (!std::is_abstract_v<T>) {
+			_GDCLASS_RECREATE(T)
```

Comment:
Now that we're no longer putting this code inside the `GDCLASS()` macro, I think I'd prefer to get rid of the `_GDCLASS_RECREATE()` macro and just inline all of that code here.